tsl/src/compression/compression.c

```diff
@@ -1450,3 +1452,53 @@ compression_get_toast_storage(CompressionAlgorithms algorithm)
 		elog(ERROR, "invalid compression algorithm %d", algorithm);
 	return definitions[algorithm].compressed_data_storage;
 }
+
+/* Get relstats from compressed chunk and insert into relstats for the
+ * corresponding uncompressed parent.
+ */
+extern void
+update_compressed_chunk_relstats(Oid uncompressed_relid, Oid compressed_relid)
+{
+	bool update = true;
+	double rowcnt;
+	int comp_pages, uncomp_pages, comp_visible, uncomp_visible;
+	float comp_tuples, uncomp_tuples, out_tuples;
+	Chunk *uncompressed_chunk = ts_chunk_get_by_relid(uncompressed_relid, true);
+	Chunk *compressed_chunk = ts_chunk_get_by_relid(compressed_relid, true);
+
+	if (uncompressed_chunk->table_id != uncompressed_relid ||
```

Comment:
Is this case possible, maybe `assert` instead?

Reply:
want this to be generic safety check here. Should not happen with the  current code path unless there is some corruption.

Reply:
Then why not assert?